### PR TITLE
EVG-5325 rest endpoint for custom notifications

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: rest/route/notification.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		})
+	}
+
+	h.composer = message.MakeJiraMessage(issue)
+	h.sender, err = h.environment.GetSender(evergreen.SenderJIRAIssue)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Error fetching sender key for evergreen.SenderJIRAIssue"))
+	}
+
+	h.sender.Send(h.composer)
```
Comment (john-m-liu) [inline rest/route/notification.go]: While this works I expected these routes to insert a notification document that would be picked up by the event_send amboy job. That way the behavior is consistent with what the app does and we can track them

File: rest/route/service.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ func AttachHandler(app *gimlet.APIApp, queue amboy.Queue, URL string, superUsers
 	app.AddRoute("/keys").Version(2).Get().Wrap(checkUser).RouteHandler(makeFetchKeys(sc))
 	app.AddRoute("/keys").Version(2).Post().Wrap(checkUser).RouteHandler(makeSetKey(sc))
 	app.AddRoute("/keys/{key_name}").Version(2).Delete().Wrap(checkUser).RouteHandler(makeDeleteKeys(sc))
+	app.AddRoute("/notifications/{target_id}").Version(2).Post().Wrap(superUser).RouteHandler(makeNotification(environment))
```
Comment (john-m-liu) [inline rest/route/service.go]: Is there anything preventing us from allowing all authenticated users to use this route?

File: rest/route/notification.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	case "jira_comment":
+		h.handler = makeJiraCommentNotification(h.environment)
+	case "jira_issue":
+		h.handler = makeJiraIssueNotification(h.environment)
+	case "slack":
+		h.handler = makeSlackNotification(h.environment)
+	case "email":
+		h.handler = makeEmailNotification(h.environment)
+	default:
+		return fmt.Errorf("'%s' is not a supported {type}", t)
```
Comment (tychoish) [inline rest/route/notification.go]: nit errors.Errorf

File: rest/route/notification.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+	}
+	email, ok := i.(*message.Email)
+	if !ok {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("Unexpected type %T for message.Email", i),
+		})
+	}
+
+	h.composer = message.MakeEmailMessage(*email)
```
Comment (tychoish) [inline rest/route/notification.go]: no pointer here? might require a grip change

Comment (StuartButler) [reply rest/route/notification.go]: *func MakeEmailMessage(e Email) Composer*

**Comment by tychoish** [comment]
fix lint? also ping @john-m-liu

**Pull request merged**